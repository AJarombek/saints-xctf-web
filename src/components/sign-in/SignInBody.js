/**
 * Container component for the sign in page.
 * @author Andrew Jarombek
 * @since 5/2/2020
 */

import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AJButton } from 'jarombek-react-components';
import ImageInput from '../shared/ImageInput';

const SignInBody = ({ signIn }) => {
  const [username, setUsername] = useState("");
  const [password, setPassword] = useState("");

  return (
    <div className="sxctf-sign-in-body">
      <div>
        <h2>Sign In</h2>
        <div>
          <ImageInput
            onChange={(e) => setUsername(e.target.value)}
            placeholder="Username or Email"
            name="username"
            type="text"
          />
          <ImageInput
            onChange={(e) => setPassword(e.target.value)}
            placeholder="Password"
            name="password"
            type="password"
          />
        </div>
        <p>Forgot Password?</p>
        <div>
          <AJButton type="contained" onClick={() => signIn(username, password)}>Sign In</AJButton>
          <AJButton type="text">Create Account</AJButton>
        </div>
      </div>
    </div>
  );
};

SignInBody.propTypes = {
  signIn: PropTypes.func.isRequired
};

export default SignInBody;
