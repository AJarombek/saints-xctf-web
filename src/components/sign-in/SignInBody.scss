/*
 * Sass styling for the SignInBody component
 * @author Andrew Jarombek
 * @since 5/2/2020
 */

@import "../../fonts";
@import "../../vars";

.sxctf-sign-in-body {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-top: 100px;
  min-height: calc(100vh - 100px);

  @media screen and (max-width: 390px) {
    margin-top: 75px;
  }

  &:before {
    content: '';
    flex-grow: 1;
    display: block;
  }

  &:after {
    content: '';
    flex-grow: 2;
    display: block;
  }

  > div {
    flex-shrink: 0;

    display: flex;
    flex-direction: column;
    align-items: center;
    position: relative;

    width: 600px;
    margin: 25px 0;

    @media screen and (max-width: 1100px) {
      width: 450px;
    }

    h2 {
      @include roboto-bold;
      font-size: 28px;
      margin: 20px 0;

      @media screen and (max-width: 1100px) {
        font-size: 24px;
      }
    }

    .sxctf-image-input {
      @media screen and (min-width: 1101px) {
        height: 60px;
        width: 325px;
      }

      img {
        @media screen and (min-width: 1101px) {
          width: 46px;
          height: 46px;
        }
      }

      input {
        @media screen and (min-width: 1101px) {
          width: 240px;
          font-size: 14px;
        }
      }
    }

    a {
      @include roboto-bold;
      font-size: 14px;
      text-decoration: none;
      align-self: flex-end;
      margin: 20px 140px 20px 0;
      color: $spot-palette-blue;

      @media screen and (max-width: 1100px) {
        margin: 20px 100px 20px 0;
        font-size: 13px;
      }

      &:hover {
        text-decoration: underline;
      }
    }

    .form-buttons {
      margin-top: 20px;

      @media screen and (max-width: 1100px) {
        margin-top: 0;
      }

      .aj-contained-button {
        margin-right: 20px;

        button {
          font-size: 14px;

          @media screen and (max-width: 1100px) {
            font-size: 11px;
          }
        }
      }

      .aj-text-button {
        button {
          font-size: 12px;

          @media screen and (max-width: 1100px) {
            font-size: 11px;
          }
        }
      }

      .aj-contained-button[class*="ajContainedButtonDisabled"] {
        background-color: #999 !important;
      }
    }
  }
}
