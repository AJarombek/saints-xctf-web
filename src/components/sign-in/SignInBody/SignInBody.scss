/*
 * Sass styling for the SignInBody component
 * @author Andrew Jarombek
 * @since 5/2/2020
 */

@import "../../../fonts";
@import "../../../vars";
@import "../../../mixins";

.sxctf-sign-in-body {
  @include centeredBody;

  > div {
    width: 600px;
    margin: 25px 0;

    @media screen and (max-width: 1100px) {
      width: 450px;
    }

    @include formTitle;

    .sxctf-image-input {
      @media screen and (min-width: 1101px) {
        height: 60px;
        width: 325px;
      }

      img {
        @media screen and (min-width: 1101px) {
          width: 46px;
          height: 46px;
        }
      }

      input {
        @media screen and (min-width: 1101px) {
          width: 240px;
          font-size: 14px;
        }
      }
    }

    @include formError;

    a {
      @include roboto-bold;
      font-size: 14px;
      text-decoration: none;
      align-self: flex-end;
      margin: 20px 140px 20px 0;
      color: $spot-palette-blue;

      @media screen and (max-width: 1100px) {
        margin: 20px 100px 20px 0;
        font-size: 13px;
      }

      &:hover {
        text-decoration: underline;
      }
    }

    .form-buttons {
      @include formButtons;

      @media screen and (max-width: 1100px) {
        margin-top: 0;
      }
    }
  }
}
