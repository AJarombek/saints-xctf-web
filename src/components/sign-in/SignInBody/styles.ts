/**
 * JSS styles for the SignInBody component.
 * @author Andrew Jarombek
 * @since 7/24/2020
 */

import Colors from '../../../styles/colors';

export default {
  contactLink: {
    margin: '0 4px !important',
    color: `${Colors.sxctfRed} !important`,
    textDecoration: 'underline !important'
  },
  errorStatus: {
    textAlign: 'center'
  }
};
