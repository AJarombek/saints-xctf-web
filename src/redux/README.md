### Overview

Configure Redux as the state management system for the application.

### Files

| Filename                 | Description                                                                |
|--------------------------|----------------------------------------------------------------------------|
| `modules`                | Redux modules containing Reducers, Actions, and Action Creators.           |
| `appState.js`            | Sample application state maintained by Redux.                              |
| `store.js`               | Configuration and creation of the Redux store.                             |
| `types.ts`               | Typescript types used throughout the application.                          |

### Resources

1) [React/Redux Example App #1](https://github.com/erikras/react-redux-universal-hot-example/tree/master/src/redux)
2) [React/Redux Example App #2](https://github.com/goopscoop/ga-react-tutorial/tree/6-reduxActionsAndReducers/src/redux)
3) [React/Redux Example App #3](https://github.com/gothinkster/react-redux-realworld-example-app/tree/master/src)
4) [Redux Ducks Spec](https://github.com/erikras/ducks-modular-redux)
5) [React Router/Redux Setup Steps](https://github.com/supasate/connected-react-router#usage)
6) [Redux Hot Reloading](https://redux.js.org/recipes/configuring-your-store/#hot-reloading)
7) [Redux Thunk](https://github.com/reduxjs/redux-thunk)
8) [Redux/React Official Real World Example](https://github.com/reduxjs/redux/tree/master/examples/real-world)
