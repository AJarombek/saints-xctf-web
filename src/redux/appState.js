/**
 * Sample Redux State for the entire application.
 * @author Andrew Jarombek
 * @since 5/3/2020
 */

const appState = {
  auth: {
    auth: {
      isFetching: false,
      lastUpdated: 1588530309,
      signedIn: true,
      status: "SUCCESS"
    },
    registration: {
      stage: 1,
      first: "Andy",
      last: "Jarombek",
      email: "andrew@jarombek.com",
      username: undefined,
      password: undefined,
      activationCode: undefined,
      teams: {
        saintsXCTFAlumni: {
          status: "accepted",
          groups: {
            alumni: {
              status: "accepted"
            }
          }
        }
      }
    },
    user: {
      andy: {
        isFetching: false,
        didInvalidate: false,
        lastUpdated: 1588530310,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        password: "xxxxxx",
        description: "Test Description",
        memberSince: "2016-02-23 12:00:00",
        classYear: 2017,
        location: "New York, NY",
        favoriteEvent: "Cooldown Jog",
        activationCode: "ABCD1234",
        email: "andrew@jarombek.com",
        lastSignIn: "2020-05-03 12:00:00",
        subscribed: "Y",
        logs: []
      }
    },
  },
  profile: {
    tom: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530311
    },
    joe: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530312
    },
    ben: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530313
    }
  },
  group: {
    wmenstf: {
      isFetching: false,
      didInvalidate: false,
      lastUpdated: 1588530314,
      groupName: "wmenstf",
      groupTitle: "Women's Track & Field",
      description: "",
      weekStart: "monday",
      members: [],
      logs: []
    }
  },
  logs: {
    isFetching: false,
    didInvalidate: false,
    lastUpdated: 1588530315,
    items: {
      1: {
        log_id: 1,
        username: "andy",
        first: "Andrew",
        last: "Jarombek",
        name: "Reservoir Loop",
        location: "New York, NY",
        date: "2020-05-03",
        type: "run",
        distance: 6.37,
        metric: "miles",
        miles: 6.37,
        time: "00:45:37",
        pace: "00:07:10",
        feel: 6,
        description: "",
        timeCreated: "2020-05-03 12:00:00",
        comments: {
          1: {
            commentId: 1,
            username: "andy",
            first: "Andrew",
            last: "Jarombek",
            logId: 1,
            time: "2020-05-03 12:00:00",
            content: "test"
          }
        }
      }
    }
  }
};
