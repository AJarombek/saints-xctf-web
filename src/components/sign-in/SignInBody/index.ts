import SignInBody from './SignInBody';

export default SignInBody;
