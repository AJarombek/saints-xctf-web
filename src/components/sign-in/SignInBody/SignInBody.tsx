/**
 * Container component for the sign in page.
 * @author Andrew Jarombek
 * @since 5/2/2020
 */

import React, { useState, useEffect } from 'react';
import { createUseStyles } from 'react-jss';
import { Link, useHistory } from 'react-router-dom';
import classNames from 'classnames';
import styles from './styles';
import { AJButton } from 'jarombek-react-components';
import ImageInput, { ImageInputStatus } from '../../shared/ImageInput';
import ImageInputSet from '../../shared/ImageInputSet';

// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
import usernameLogo from '../../../../assets/username.png';

// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
// @ts-ignore
import passwordLogo from '../../../../assets/password.png';

interface Props {
  signIn: Function;
  isFetching?: boolean;
  status?: string;
}

const useStyles = createUseStyles(styles);

const SignInBody: React.FunctionComponent<Props> = ({ signIn, isFetching, status }) => {
  const classes = useStyles();

  const history = useHistory();

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [loading, setLoading] = useState(false);
  const [iconStatus, setIconStatus] = useState<ImageInputStatus>(ImageInputStatus.NONE);
  const [errorStatus, setErrorStatus] = useState(null);

  useEffect(() => {
    let message;
    switch (status) {
      case 'INVALID PASSWORD':
      case 'INVALID USER':
        message = 'Invalid username and password combination.';
        setErrorStatus(message);
        setIconStatus(ImageInputStatus.FAILURE);
        break;
      case 'INTERNAL ERROR':
        message = (
          <>
            An unexpected error occurred. Contact
            <a className={classes.contactLink} href="mailto:andrew@jarombek.com">
              andrew@jarombek.com
            </a>
            if this error persists.
          </>
        );
        setErrorStatus(message);
        setIconStatus(ImageInputStatus.FAILURE);
        break;
      default:
        setErrorStatus(null);
        setIconStatus(ImageInputStatus.NONE);
    }
  }, [status, classes.contactLink]);

  const onClickSignIn = async (): Promise<void> => {
    if (username.length !== 0 && password.length !== 0) {
      setLoading(true);
      await signIn(username, password);
      setLoading(false);
    }
  };

  return (
    <div className="sxctf-sign-in-body">
      <div>
        <h2>Sign In</h2>
        <div>
          <ImageInputSet>
            <ImageInput
              onChange={(e): void => setUsername(e.target.value)}
              icon={usernameLogo}
              placeholder="Username or Email"
              name="username"
              type="text"
              autoComplete="username"
              status={iconStatus}
            />
            <ImageInput
              onChange={(e): void => setPassword(e.target.value)}
              icon={passwordLogo}
              placeholder="Password"
              name="password"
              type="password"
              autoComplete="current-password"
              status={iconStatus}
            />
          </ImageInputSet>
        </div>
        {errorStatus && <p className={classNames(classes.errorStatus, 'errorStatus')}>{errorStatus}</p>}
        <Link to="/forgotpassword">Forgot Password?</Link>
        <div className="form-buttons">
          <AJButton type="contained" onClick={onClickSignIn} disabled={isFetching || loading}>
            Sign In
          </AJButton>
          <AJButton type="text" onClick={(): void => history.push('/register')}>
            Create Account
          </AJButton>
        </div>
      </div>
    </div>
  );
};

export default SignInBody;
