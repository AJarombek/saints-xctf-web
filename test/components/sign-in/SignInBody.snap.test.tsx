/**
 * Snapshot test for the SignInBody component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { Store } from 'redux';
import { emptyStore } from '../../test-utils/storeMocks';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter } from 'react-router-dom';
import SignInBody from '../../../src/components/sign-in/SignInBody';

const mockStore = configureStore([thunk]);

describe('SignInBody Snapshot Tests', () => {
  let store: Store;

  beforeEach(() => {
    store = mockStore(emptyStore);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <SignInBody isFetching={false} status="" />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
