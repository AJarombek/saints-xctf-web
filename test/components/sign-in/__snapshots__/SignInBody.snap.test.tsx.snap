// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInBody Snapshot Tests renders correctly 1`] = `
<div
  className="sxctf-sign-in-body"
>
  <div>
    <h2>
      Sign In
    </h2>
    <div>
      <div
        className="sxctf-image-input-set row"
      >
        <form>
          <div
            className="sxctf-image-input"
            data-cypress="image-input-username"
          >
            <img
              alt=""
              src="test-file-stub"
            />
            <input
              autoComplete="username"
              name="username"
              onChange={[Function]}
              placeholder="Username or Email"
              type="text"
            />
            <div>
              <div
                className="status none"
              >
                <p />
              </div>
            </div>
          </div>
          <div
            className="sxctf-image-input"
            data-cypress="image-input-password"
          >
            <img
              alt=""
              src="test-file-stub"
            />
            <input
              autoComplete="current-password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
            />
            <div>
              <div
                className="status none"
              >
                <p />
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <a
      href="/forgotpassword"
      onClick={[Function]}
    >
      Forgot Password?
    </a>
    <div
      className="form-buttons"
    >
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-contained-button ajContainedButton-2-2-1 ajContainedButtonEnabled-2-2-2"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Sign In
        </button>
      </div>
      <div
        className="ajBaseButton-2-2-4 ajBaseButtonEnabled-2-2-5 aj-text-button ajTextButton-2-2-7 ajTextButtonEnabled-2-2-8"
        onClick={[Function]}
      >
        <button
          disabled={false}
          type="button"
        >
          Create Account
        </button>
      </div>
    </div>
  </div>
</div>
`;
